keps/sig-scheduling/4832-async-preemption/README.md

```diff
@@ -0,0 +1,999 @@
+-->
+
+The scheduler is basically only one in a cluster,
+and hence scheduling throughput is the crutial metric for the scheduler.
+
+The scheduler schedules Pods one by one within the scheduling cycle, 
+and we basically try to reduce the API calls as much as possible to enhance the scheduling cycle throughput.
+
+The binding cycle is the example for this motivation; 
+1. The scheduling cycle decides where Pod should go to,
+2. At the end of the scheduling cycle, the scheduler reserves the Node within the scheduler's cache so that next scheduling cycle will take the current pod into consideration.
+3. The scheduling cycle ends and the binding cycle starts; the binding cycle is run asynchronously, and the scheduler starts the next scheduling cycle.
+
+This flow allows us to decouple the API call to assign Pod to the Node from the scheduling cycle so that the API call doesn't block the scheduling throughput.
+
+But, we have the similar problem with the preemption; the preemption is run at PostFilter extension point which is the part of the scheduling cycle.
+The preemption has to make some API calls to update Pods' condition and delete Pods after all, which could block the scheduling throughput.
+
+scheduler-perf [actually shows](https://github.com/kubernetes/kubernetes/blob/342da505bdefbd849b808cca3cb76c24a993025f/test/integration/scheduler_perf/config/performance-config.yaml#L641) currently the preemption scenario takes too long time, compared to others.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- The preemption plugin makes API calls for the preemption asynchronously after `PostFilter` extension point so that the scheduler can continue to other Pods' scheduling while making API calls for preemption.
+  - Until the preemption goroutine is done, the Pod won't be rescheduled.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Making the same enhancement for DRA is not a goal of this KEP because it's an under-construction feature yet.
+  - If DRA maintainers want, technically they can along with this KEP. But, at least in this KEP, we don't discuss how.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
```

Comment:
This story doesn't seem to match the proposal anymore

Reply:
Will just remove the story part as we only focus on optimizing our preemption, not like creating a new extension point, which could have multiple user stories.